.scalafix.conf

```diff
@@ -18,9 +18,66 @@ SortImports.blocks = [
   "scala."
 ]
 
+Disable.symbols = [
+  {
+    regex = "^\\Qscala/collection/mutable\\E.*$"
+    message = "Java collections have better performance, which is what I assume you were trying to do"
+  }
+  {
+    regex = "^\\Qscala/collection/parallel\\E.*$"
+    message = "parallel collections are low performance and block the thread, prefer cats.parMapN"
+  }
+  {
+    regex = "^\\Qscala/math/Big\\E.*$"
+    message = "scala arbitrary precision numbers are broken: https://github.com/scala/bug/issues/9670"
+  }
+  {
+    regex = {
+      includes = [
+        "^\\Qjava/io\\E.*$"
```

Comment:
> need to use `java.io` for compatibility reasons
Would you say those are exceptional reasons? (For example, in current SST, it was used only in the `Console` module, which is more like a helper utility than a production grade thing anyway.)

Reply:
In some way yes. But I do think that using `java.io` is just fine - you just need to be careful about blocking - but you have to do that anyway with lots of other code. Benefits of `java.io` are that everyone knows them and they are quite easy to use. On the other hand I don't think anyone would use `java.nio` directly, as far as I know that's not really possible so you would always use something more high-level (such as Netty). My point is that you either use some higher-level library or you use `java.io` directly because you need to.

Reply:
then I would say that those rare exceptional cases deserve a `@SuppressWarnings(Array("scalafix:Disable...."))`, don't you think?

Reply:
No, it would be exceptional if I would be using `java.nio` all the time. Often I use `InputStream` or `OutputStream` (for whatever reason) and I don't want to be forced to suppres warnings everywhere because I belive `java.io` is perfectly fine to be used. We should be banning things that are harmful, `java.io` is not harmful.

Reply:
> `java.io` is perfectly fine to be used ... `java.io` is not harmful

From scalazzi point of view (and FP in general), that's exactly what it is, harmful. It's side-effectful _and_ blocking. Now of course, I don't see this as a moral issue, it's just that mixing cats-effect stuff with `java.io` is bloody tricky and scalafix could help us with avoiding that :smile_cat:
Are you really sure you want SST code (written by you or any other contributor) to use `InputStream` or `OutputStream` willy-nilly?

Reply:
Yes, I am sure.

@tomasherman @hanny24 Opinions?

Reply:
I think that disabling `java.io` is very radical. Sure, we should prefer `java.nio` where it makes sense. I think we should not try to make our build very strcit as it leads to a lot of exceptions (`SuppressWarnings`)...We can limit the use of `java.io` by the means of PRs.

Reply:
I'm not a fan of all these unnecessary restrictions - in my opinion time will be better spent on actually adding functionality